### DISCUSSION TITLE
Check for rendering elements

### DISCUSSION THREAD

**Discussion by VedankNaik** [iTwin/itwinjs-core#2864] [Q&A] [answered]
Hello, 
I need to display a loading page until all the elements are rendered in the viewer. Is there anything that can be used to do the same?

**Comment by pmconne** [comment, marked as answer]
Hi @VedankNaik.
The viewer displays tiles, not elements. [This React component](https://www.itwinjs.org/reference/ui-framework/statusbar/tileloadingindicator/) displays a progress bar indicating what percentage of tiles required by the current view have been loaded. Hope this helps - if not please clarify.

**Reply by VedankNaik** [reply]
Can you tell me what can I do to show/hide my loading component while tiles are getting loaded?

**Reply by aruniverse** [reply]
The itwin viewer adds the above component mentioned by paul by default to the status bar. It'll auto show/hide depending on if tiles are being fetched. 
![image](https://user-images.githubusercontent.com/11051042/145233204-754d996c-d674-4f1a-a8e2-48d851092914.png)

**Reply by pmconne** [reply]
By "my loading component" are you referring to the TileLoadingIndicator I linked, or some other component?
If the latter, I cannot assist you - I am not a UI developer.
If the former, I believe it automatically hides itself when no tiles are loading - @bsteinbk?

**Reply by bsteinbk** [reply]
The TileLoading Indicator does hide itself once it detects that no more tiles are to be loaded. It hides the component by [adding a class](https://github.com/iTwin/itwinjs-core/blob/d60ad65c1227d071737b66211a65897b43b543c3/ui/appui-react/src/appui-react/statusfields/tileloading/TileLoadingIndicator.tsx#L114) to the DOM Element.

**Reply by VedankNaik** [reply]
I was able to solve my problem by using logic similar to what is used in TileLoadingIndicator to show/hide component. Thanks
